Homework-1.0

anenvs opened the pull request:
Описание задания:

В ресурсах хранятся вопросы и различные ответы к ним в виде CSV файла (5 вопросов).
Вопросы могут быть с выбором из нескольких вариантов или со свободным ответом - на Ваше желание и усмотрение.
Приложение должна просто вывести вопросы теста из CSV-файла с возможными вариантами ответа (если имеются).


Требования:
В приложении должна присутствовать объектная модель (отдаём предпочтение объектам и классам, а не строчкам и массивам/спискам строчек).
Все классы в приложении должны решать строго определённую задачу (см. п. 18-19 "Правила оформления кода.pdf", прикреплённые к материалам занятия).
Контекст описывается XML-файлом.
Все зависимости должны быть настроены в IoC контейнере.
Имя ресурса с вопросами (CSV-файла) необходимо захардкодить строчкой в XML-файле с контекстом.
CSV с вопросами читается именно как ресурс, а не как файл.
Scanner, PrintStream и другие стандартные типы в контекст класть не нужно!
Весь ввод-вывод осуществляется на английском языке.
Крайне желательно написать юнит-тест какого-нибудь сервиса (оцениваться будет только попытка написать тест).

anenvs committed:
Homework-1.0

vlad-iv commented on pom.xml:
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Предлагаю для проверки сборки, кодстайла, тестов проекта на гитхабе
создать файл .github/workflows/build.yml
Пример на Java 17, можно заменить на 21:
https://github.com/OtusTeam/Spring/blob/master/.github/workflows/build.yml

vlad-iv commented on pom.xml:
```diff
@@ -0,0 +1,112 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
```
Можно добавить корневой pom.xml
где модулями будут ДЗ (названия папок , hw01 и тд. в следующих ДЗ)
пример
https://github.com/OtusTeam/Spring/blob/master/2025-01/spring-01/pom.xml

И перенести все файлы текущей работы в папку (модуль) hw01

vlad-iv commented on src/main/java/ru/sidorov/dao/CsvQuestionDao.java:
```diff
@@ -0,0 +1,39 @@
+        try (
+                InputStream inputStream = getClass().getClassLoader()
+                        .getResourceAsStream(fileNameProvider.getTestFileName());
+                InputStreamReader reader = new InputStreamReader(inputStream)) {
```
Предлагаю проверить inputStream  сначала на null
можно через Objects.requiredNotNull
если null то  в сообщении можно указать, что ресурс (путь к нему) не найден

vlad-iv commented on src/main/resources/spring-context.xml:
```diff
@@ -0,0 +1,31 @@
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+        http://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.springframework.org/schema/util https://www.springframework.org/schema/util/spring-util.xsd">
+
```
Предлагаю добавить тесты на CsvQuestionDao
1. проверить, что из тестового ресурса загрузить вопросы и ответ
2. проверить, что будет исключение QuestionReadException , если ресурс не найден

anenvs committed:
Homework-1.1
1. Added CsvQuestionDaoTest.class
2. Added github/workflows/build.yml
3. Added requireNonNull inputStream check at CsvQuestionDao
4. Project splitted on modules

anenvs committed:
Homework-1.2
1. CsvQuestionDaoTest.findAll_fileExists fixed
